torchao/float8/float8_linear.py

```diff
@@ -395,13 +531,33 @@ def forward(self, input: torch.Tensor) -> torch.Tensor:
         if self.has_any_delayed_scaling:
             self.float8_pre_forward(input)
 
-        input_fp8 = self.cast_input_to_float8(input, self.is_amax_initialized)
-        weight_fp8 = self.cast_weight_to_float8(self.weight, self.is_amax_initialized)
+        # TODO(this PR): reuse with config, make a property
+        has_all_axiswise_scaling = (
+            self.config.cast_config_input.scaling_granularity is ScalingGranularity.AXISWISE and
+            self.config.cast_config_weight.scaling_granularity is ScalingGranularity.AXISWISE and
+            self.config.cast_config_grad_output.scaling_granularity is ScalingGranularity.AXISWISE
+        )
+
+        if not has_all_axiswise_scaling:
+            input_fp8 = self.cast_input_to_float8(input, self.is_amax_initialized)
+            weight_fp8 = self.cast_weight_to_float8(self.weight, self.is_amax_initialized)
 
-        output = manual_float8_matmul.apply(input_fp8, weight_fp8.t())
+            output = manual_float8_matmul_with_args_in_float8.apply(input_fp8, weight_fp8.t())
 
-        # Cast grad_output to float8_e5m2 during backward
-        output = self.cast_output_to_float8_in_bw(output)
+            # Cast grad_output to float8_e5m2 during backward
+            output = self.cast_output_to_float8_in_bw(output)
+
+        else:
```

Comment:
why do we need a separate path for this?

Reply:
this is choosing to move fast on a single use case at the expense of taking on some temporary tech debt